src/Dashboard/Dashboard.php

```diff
@@ -363,56 +363,52 @@ public function cloneCurrent(): array
      *
      * @param bool   $force don't check dashboard are already loaded and force their load
      * @param bool   $check_rights use to remove rights checking (use in embed)
-     * @param string $context only dashboard for given context
+     * @param ?string $context only dashboard for given context
      *
      * @return array dasboards
      */
-    public static function getAll(
-        bool $force = false,
-        bool $check_rights = true,
-        string $context = 'core'
-    ): array {
+    public static function getAll(bool $force = false, bool $check_rights = true, ?string $context = 'core'): array
+    {
         global $DB;
 
-        if (!$force && count(self::$all_dashboards) > 0) {
-            return self::$all_dashboards;
-        }
-
-       // empty previous data
-        self::$all_dashboards = [];
-
-        $dashboard_criteria = [];
-        if (strlen($context)) {
-            $dashboard_criteria['context'] = $context;
-        }
-
-        $dashboards = iterator_to_array($DB->request(self::getTable(), ['WHERE' => $dashboard_criteria]));
-        $items      = iterator_to_array($DB->request(Item::getTable()));
-        $rights     = iterator_to_array($DB->request(Right::getTable()));
-
-        foreach ($dashboards as $dashboard) {
-            $key = $dashboard['key'];
-            $id  = $dashboard['id'];
-
-            $d_rights = array_filter($rights, function ($right_line) use ($id) {
-                return $right_line['dashboards_dashboards_id'] == $id;
-            });
-            $dashboardItem = new self($key);
-            if ($check_rights && !$dashboardItem->canViewCurrent()) {
-                continue;
+        if ($force || count(self::$all_dashboards) == 0) {
+            self::$all_dashboards = [];
+
+            $dashboards = iterator_to_array($DB->request(self::getTable()));
+            $items      = iterator_to_array($DB->request(Item::getTable()));
+            $rights     = iterator_to_array($DB->request(Right::getTable()));
+
+            foreach ($dashboards as $dashboard) {
+                $key = $dashboard['key'];
+                $id  = $dashboard['id'];
+
+                $d_rights = array_filter($rights, static function ($right_line) use ($id) {
```

Comment:
What's the purpose of static anonymous function here (and below) ?

Reply:
Same as old code but the use of the "static" keyword on anonymous functions is preferred if you don't need access to properties or methods of the object instance. It is probably just a minor performance improvement.

Reply:
ok